### DISCUSSION TITLE
simple inheritance

### DISCUSSION THREAD

**Discussion by robaho** [SwiftyLab/MetaCodable#112] [Q&A] [answered]
Is there any way to do this?

```
@Codable
class A {
   var name : String
}
@Codable
class B : A {
   var location : String
}

@Codable
class Container {
    var elements[A] = [ B() ]
}
```

How do I annotate this so that when Container is decoded the correct instances of A/B are instantiated?

I got dynamic coding with a protocol working but I'd rather not use that. It seems it only works with protocols and enums?

I could code it by hand to work, but it seems the purpose of MetaCodable is to make something like this easy.

**Comment by soumyamahunt** [comment, marked as answer]
> I got dynamic coding with a protocol working but I'd rather not use that. It seems it only works with protocols and enums?

@robaho I am assuming you are talking about `MetaProtocolCodable`, you can use enums without having to use the `MetaProtocolCodable`. `MetaCodable` by itself provides [untagged](https://swiftpackageindex.com/swiftylab/metacodable/v1.4.0/documentation/metacodable/untagged()), external, internal and adjacent tagging enum decoding functionality. For your use case, it could look something like:

```
@Codable
struct Container
  let elements: [AOrB]

  @Codable
  @UnTagged
  enum AOrB {
    case b(name: String, location : String)
    case a(name: String)
  }
}
```

Please use discussion to ask these kind of queries. And use the proper templates when creating issues.

**Reply by robaho** [reply]
Sorry. I will use discussions in the future. I was able to resolve this using regular enums. I will post back when I get a moment.
